src/build/env.py

```diff
@@ -132,14 +154,15 @@ def install(self, requirements):  # type: (Iterable[str]) -> None
                 '-r',
                 os.path.abspath(req_file.name),
             ]
-            subprocess.check_call(cmd)
+            subprocess.check_call(cmd, stdout=self._outfd)
```

Comment:
What about stderr?

Reply:
Well, if there is an error we want to report that. All the things we are using seem to follow the standard of only printing to stderr when things go wrong.

Reply:
> All the things we are using seem to follow the standard of only printing to stderr when things go wrong.

Can you detail what things are you referring to?

Reply:
Okay, so, all these changes about the stdout handler are to private API. I introduced a `silent` bool argument to `IsolatedEnvBuilder`, but for easier implementation I opted to pass a file descriptor for stdout to most of the private API.
In the private API, remember this is the environment building, we only call pip, ensurepip and virtualenv, and all of those will only print to stderr when something goes wrong. This means if everything goes fine, there will be no output, but if there is an error in the subprocess they can still print to stderr and the users will see the error.

TLDR: I don't want to swallow stderr because I want users to see the error if for eg. pip fails install a bad dependency. Only stdout, which irrelevant if everything goes right.

Does that make sense?

Reply:
@gaborbernat does this address your concerns? If so, I can go ahead and push a new version fixing everything.

Reply:
My preferred approach to this problem is to always swallow everything in a silent mode, but save in memory. If the build fails show all. This is what tox does too. Sometimes the stdout also contains useful information to track down issues.